stl/inc/mutex

```diff
@@ -110,23 +93,16 @@ private:
     friend condition_variable;
     friend condition_variable_any;
 
-    union {
-        _Aligned_storage_t<_Mtx_internal_imp_size, _Mtx_internal_imp_alignment> _Mtx_storage;
-        _Mtx_internal_imp_mirror _Mtx_storage_mirror;
-    };
+    _Mtx_internal_imp_t _Mtx_storage{};

```

Comment:
This will break the debugger definitions; we probably want that to be updated as part of this.

Reply:
Oh, yep, we discussed this in the other PR, nvm.